x-pack/plugins/beats/index.js

```diff
@@ -4,13 +4,22 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { registerApiRoutes } from './server/routes/api';
 import { PLUGIN } from './common/constants';
 
+const HOURS_IN_SECONDS = 60 * 60;
+
 export function beats(kibana)  {
   return new kibana.Plugin({
     id: PLUGIN.ID,
     require: ['kibana', 'elasticsearch', 'xpack_main'],
-    init: function () {
+    configPrefix: 'xpack.beats',
+    config: Joi => Joi.object({
+      enabled: Joi.boolean().default(true),
+      enrollmentTokensTtlInSeconds: Joi.number().integer().min(1).default(4 * HOURS_IN_SECONDS)
```

Comment:
@exekias WDYT about the default value of 4 hours for enrollment token expiration? Users (admins) could always override it via the `xpack.beats.enrollmentTokensTtlInSeconds` setting in `kibana.yml`.

Reply:
Related question: do you think this value should be override-able per enrollment token? This would mean that the `POST /api/beats/enrollment_tokens` API would take an optional parameter for the expiration timestamp.

Reply:
About enrollment token expiration, I think TTLs must be low, something like 10 mins, for instance, to stress the idea of "this is single time use, use it right away". We can play with these values later.

As for the override-able feature, it's interesting, I would be ok with having that at API level, I don't see a fit for it in the UI, but we can discuss that too when we go over new mockups.

Reply:
>About enrollment token expiration, I think TTLs must be low, something like 10 mins, for instance, to stress the idea of "this is single time use, use it right away". We can play with these values later. 

Thanks, I will change the default to 10 minutes.

> As for the override-able feature, it's interesting, I would be ok with having that at API level, I don't see a fit for it in the UI, but we can discuss that too when we go over new mockups.

I think my question is more along the lines of this: we will provide a default of 10 minutes. Users can override this default by changing the value of `xpack.beats.enrollmentTokensTtlInSeconds` in `kibana.yml`. Overriding this setting will affect _all_ enrollment tokens created from that point forward. Is this level of override-ability enough? Or do we additionally want users to be able to provide different expiration times _per enrollment token at runtime_ as well, via the "create enrollment token" API?

Reply:
I don't have a strong opinion about the parameter, it may be useful in some use cases, I'm ok to add it if it's easy

Reply:
Thanks. In that case I'm going to suggest we *not* add it to the API just yet. Let's wait till we have an actual use case for it and we can easily add it then. My preference is to keep the API minimal until we need to grow it.

Reply:
Should we give a constant name to the value you're supplying here?